src/components/common/dialog.tsx

```diff
@@ -0,0 +1,789 @@
+                this.update();
+                if (this.online_validation)
+                    this.trigger_validation();
+            },
+            String(tag));
+    }
+
+    async validate(): Promise<boolean> {
+        this.trigger_validation();
+        // TODO - wait for the asyncs to be done
+        return Object.keys(this.validation).length == 0;
+    }
+}
+
+export function useDialogState<V extends object>(
+    init: V | (() => V),
+    validate: (dlg: DialogState<V>) => void
+) : DialogState<V> {
+    const dlg = useObject(() => {
+        if (typeof init == "function")
+            return new DialogState(init(), validate);
+        else
+            return new DialogState(init, validate);
+    }, null, []);
+    useOn(dlg, "changed");
+    return dlg;
+}
+
+// export function useDialogState_async<V>(...)
+
+// Common elements
+
+export class DialogError {
+    title: string;
+    details: React.ReactNode;
+
+    constructor(title: string, details: React.ReactNode) {
+        this.title = title;
+        this.details = details;
+    }
+
+    toString() {
+        return this.title + ": " + String(this.details);
+    }
+
+    static fromError(title: string, err: unknown) {
+        if (err && typeof err == "object" && "message" in err && typeof err.message == "string")
+            return new DialogError(title, err.message);
+        else
+            return new DialogError(title, String(err));
+    }
+}
+
+export function DialogErrorMessage<V>({
+    dialog,
+} : {
+    dialog: DialogState<V>
+}) {
+    if (!dialog.error)
+        return null;
+
+    let title: string;
+    let details: React.ReactNode;
+
+    const err = dialog.error;
+    if (err instanceof DialogError) {
+        title = err.title;
+        details = err.details;
+    } else if (err && typeof err == "object" && "message" in err && typeof err.message == "string") {
+        title = _("Failed");
+        details = err.message;
+    } else {
+        title = _("Failed");
+        details = String(err);
+    }
+
+    return (
+        <Alert
+            id="dialog-error-message"
+            variant='danger'
+            isInline
+            title={title}
+        >
+            {details}
+        </Alert>
+    );
+}
+
+export function DialogActionButton<V>({
+    dialog,
+    children,
+    action,
+    onClose = undefined,
+    ...props
+} : {
+    dialog: DialogState<V> | null,
+    children: React.ReactNode,
+    action: (values: V) => Promise<void>,
+    onClose?: undefined | (() => void)
+} & Omit<ButtonProps, "id" | "action" | "isLoading" | "isDisabled" | "variant" | "onClick">) {
+    return (
+        <Button
+            id="dialog-apply"
+            isLoading={!!dialog && dialog.busy}
+            isDisabled={!dialog || dialog.actions_disabled}
+            variant="primary"
+            onClick={() => {
+                cockpit.assert(dialog);
+                dialog.run_action(vals => action(vals).then(onClose || (() => {})));
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2425-899c38e3-20251201-074921-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)